### PR TITLE
add landing page for rust telemetry workshop

### LINKED ISSUE #2171: Create landing page for Rust telemetry workshop
We want to offer [this workshop](https://mainmatter.com/services/workshops/telemetry-for-rust-apis/) again in December, as an online event that folks can sign up for.

We need a landing page for it which includes the datetime information as well as a button to book a seat (which will be handled via ti.to cc @marcoow).

Happy to help with the content/any clarification required!

### PR DESCRIPTION
**Pull request by yaigrie**
closes #2171

### CODE REVIEW

**Comment by yaigrie** [comment]
@LukeMathWalker the landing page for your workshop :)

File: src/rust-api-telemetry-workshop.njk [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+          </div>
+        </div>
+      </div>
+
+      <section class="lp-section lang-features container container--lg">
+        <p>
+        Your Rust application has finally been deployed to production! Nice! But is
+        it working? This workshop will introduce you to a comprehensive toolkit to
+        detect, troubleshoot and resolve issues in your Rust APIs.
+        </p>
```
Comment (LukeMathWalker) [inline src/rust-api-telemetry-workshop.njk]:
```suggestion
        <p>
        Your Rust application has finally been deployed to production! Nice! But is
        it working?<br> 
        This workshop will introduce you to a comprehensive toolkit to
        detect, troubleshoot and resolve issues in your Rust applications.
        </p>
```

File: src/rust-api-telemetry-workshop.njk [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+          "loading": "lazy",
+          "sizesArray": [760, 1440, 1920]
+        } %}
+        {{ imageAspectRatio(imageData, "32/13", "35/19") }}
+      </section>
+
+      <section class="lp-section container container--lg">
+        <h2 class="lp-section-subheading h2">Additional Information</h2>
+        <h3 class="h4">Duration & location</h3>
+        <p>The workshop takes place over two afternoons on Dec. 12th, and 13th., 14:00 to 18:00 CET each. That allows participants to use the mornings to exercise or keep up with work.
```
Comment (LukeMathWalker) [inline src/rust-api-telemetry-workshop.njk]: Quick way for people to check the time in their local timezone
```suggestion
        <p>The workshop takes place over two afternoons on Dec. 12th, and 13th., <a href="https://everytimezone.com/s/03cf783a">14:00 to 18:00 CET each</a>. That allows participants to use the mornings to exercise or keep up with work.
```

File: src/rust-api-telemetry-workshop.njk [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+            The workshop is run completely online and remote.</p>
+      </section>
+      <section class="lp-section container container--lg">
+        <h3 class="h4">Number of participants</h3>
+        <p>TTo ensure the highest quality, we accept up to 15 participants.</p>
```
Comment (LukeMathWalker) [inline src/rust-api-telemetry-workshop.njk]:
```suggestion
        <p>To ensure the highest quality, we accept up to 15 participants.</p>
```

File: src/rust-api-telemetry-workshop.njk [hunk 4]
```diff
@@ -0,0 +1,238 @@
...
+      </section>
+
+      <section class="lp-section container container--lg">
+        {% set 'content' = {
+          "text": "Excellent workshop! It should be noted that it requires a solid base level of Rust knowledge [...]. Here I was able to fully use the time and knowledge of the trainer! 👍",
+          "source": "Attendee of Luca's workshop held at EuroRust 2023",
+          "loading": "lazy"
+        } %}
+        {{ quote(content) }}
+      </section>
```
Comment (LukeMathWalker) [inline src/rust-api-telemetry-workshop.njk]: Let's remove this and consolidate everything in the first quote.

File: src/rust-api-telemetry-workshop.njk [hunk 5]
```diff
@@ -0,0 +1,238 @@
...
+        <p>
+        This workshop
+        is designed for developers who are operating Rust services in production-like
+        environments, or are preparing to do so.
+        </p>
+      </section>
+
+      <section class="lp-section container container--lg">
+        {% set 'content' = {
+  "text": "I usually avoid workshops. The pace of more than 3 people is hard to sync: I always feel rushed in some bits, slowed down in others. You fixed that with the code-centric approach, I really enjoyed that.",
```
Comment (LukeMathWalker) [inline src/rust-api-telemetry-workshop.njk]:
```suggestion
  "text": "I usually avoid workshops. The pace of more than 3 people is hard to sync: I always feel rushed in some bits, slowed down in others. Luca fixed that with his code-centric approach [...] Here I was able to fully use the time and the knowledge of the trainer!",
```

File: src/rust-api-telemetry-workshop.njk [hunk 6]
```diff
@@ -0,0 +1,238 @@
+---
+title: "Workshop: You can't fix what you can't see: telemetry for Rust APIs"
```
Comment (LukeMathWalker) [inline src/rust-api-telemetry-workshop.njk]:
```suggestion
title: "Workshop: You can't fix what you can't see: telemetry for Rust applications"
```

File: src/rust-api-telemetry-workshop.njk [hunk 7]
```diff
@@ -0,0 +1,238 @@
...
+description: "Your Rust application has finally been deployed to production! Nice! But is
+  it working? This workshop will introduce you to a comprehensive toolkit to
+  detect, troubleshoot and resolve issues in your Rust APIs. This workshop
```
Comment (LukeMathWalker) [inline src/rust-api-telemetry-workshop.njk]:
```suggestion
  detect, troubleshoot and resolve issues in your Rust applications. This workshop
```

File: src/rust-api-telemetry-workshop.njk [hunk 8]
```diff
@@ -0,0 +1,238 @@
...
+      <section class="color-hero color-hero--purple" data-background-color="purple">
+        <div class="color-hero__image-wrapper">
+          {% image '/assets/images/workshops/telemetry-for-rust-apis/header-background.jpg', '2 people in front of a notebook with 3 charts on the screen, only their arms visible, one points at the screen', '100vw', 'eager', 'color-hero__image', [760, 1440, 1920] %}
+        </div>
+        <div class="color-hero__overlay" data-color-hero></div>
+        <div class="color-hero__container container container--lg color-hero__container--center workshop-lp-hero__container">
+          <div class="color-hero__main">
+            <img class="workshop-lp-hero__logo" src="/assets/images/web-based-services-in-rust-workshop/rust-logo.svg" alt="Rust logo"/>
+            <p class="workshop-hero__format">Remote Workshop</p>
+            <h1 class="workshop-lp-hero__title color-hero__title color-hero__title--full">Telemetry for Rust APIs</h1>
```
Comment (LukeMathWalker) [inline src/rust-api-telemetry-workshop.njk]:
```suggestion
            <h1 class="workshop-lp-hero__title color-hero__title color-hero__title--full">Telemetry for Rust applications</h1>
```

File: src/rust-api-telemetry-workshop.njk [hunk 9]
```diff
@@ -0,0 +1,238 @@
...
+
+      <section class="color-hero color-hero--purple" data-background-color="purple">
+        <div class="color-hero__image-wrapper">
+          {% image '/assets/images/web-based-services-in-rust-workshop/cta-background.jpg', "a pair of hands tie an entry band around someone's arm", '100vw', 'lazy', 'color-hero__image', [760, 1440, 1920] %}
+        </div>
+        <div class="color-hero__overlay"></div>
+        <div class="color-hero__container container container--lg color-hero__container--center">
+          <div class="color-hero__main">
+            <h1 class="color-hero__title color-hero__title--full">Book your ticket</h1>
+            <div class="color-hero__text color-hero__text--small">Join our workshop and start detect, troubleshoot and resolve issues in your Rust APIs.</div>
```
Comment (LukeMathWalker) [inline src/rust-api-telemetry-workshop.njk]:
```suggestion
            <div class="color-hero__text color-hero__text--small">Join our workshop and start detect, troubleshoot and resolve issues in your Rust applications.</div>
```

**Comment by LukeMathWalker** [comment]
I've decided to rebrand it to "Telemetry for Rust applications", rather than APIs. I think I've fixed all occurrences in the copy 👀

File: src/rust-api-telemetry-workshop.njk [hunk 10]
```diff
@@ -0,0 +1,238 @@
...
+        "color": "black",
+        "tag": "Workshop Overview - Part 3",
+        "title": "Panic handling",
+        "text": "You should always manage to capture details about what went wrong, even if it’s due to an uncaught panic rather than an error. We will review panic hooks and integrate them in our tracing setup."
+      } %}
+
+      {% set 'slide_4' = {
+        "color": "aqua",
+        "tag": "Workshop Overview - Part 4",
+        "title": "Testing",
```
Comment (LukeMathWalker) [inline src/rust-api-telemetry-workshop.njk]:
```suggestion
        "title": "Metrics",
```

File: src/rust-api-telemetry-workshop.njk [hunk 11]
```diff
@@ -0,0 +1,238 @@
...
+          "imgPath": "/assets/images/rust-telemetry-workshop/rust-telemetry-workshop-eurorust-2023.jpg",
+          "alt": "Happy workshop attendee smiling into the camera",
+          "sizes": "100vw",
+          "loading": "lazy",
+          "sizesArray": [760, 1440, 1920]
+        } %}
+        {{ imageAspectRatio(imageData, "32/13", "35/19") }}
+      </section>
+
+      {% set 'slide_1' = {
```
Comment (LukeMathWalker) [inline src/rust-api-telemetry-workshop.njk]: I find the slide transitions really difficult to read—I'm used to scroll as I go through a page and with these slides the content disappears before I get to it.
Any chance we could have them as "plain" sections?

**Comment by LukeMathWalker** [comment]
It feels like the "call to action" buttons are too hidden.
The one at the top:
image
disappears as soon as you start scrolling through the page, which happens pretty much immediately since there is no content beyond the title in the visible viewport when you land on the page.
The other one is hidden at the very bottom.

Can we structure it in such a way that the button is more visible, especially when you get to the "additional information"?
My preference would be for a sticky header, which makes it so that you always see it when you are going through the page, but I'm open to any kind of solution.

**Comment by LukeMathWalker** [comment]
Do we need all this empty space between the title and the beginning of the content?
image
We have a lot of vertical whitespace in a few other places as well—e.g. here after the abstract and before the quote:
image

**Comment by yaigrie** [comment]
@LukeMathWalker thank you for your great in depth feedback! It's very helpful
I will try to implement your suggestions :)

**Comment by marcoow** [comment]
@yaigrie not directly part of the work on this landing page but we'll also need a header image for this workshop like the one we have on the Svelte workshop ti.to page: https://ti.to/mainmatter/svelte-sveltekit-workshop

**Comment by yaigrie** [comment]
@LukeMathWalker 
Regarding the Navigation: Maybe we should consider a site-wide sticky navigation, because I feel like it would be weird to only have a sticky nav on that particular workshop page. As a quick solution I added a CTA button right below the intro text. 
But I think it would be nice to have a global sticky navigation. What do you think @marcoow?
Of course the Logo and the Button would have to be adjusted depending on the background colour, so that may take me a while to figure a nice solution

Regarding the Slides: 
I think this is also a problem that affects not only this workshop page, but the whole site. So maybe we could find a solution for that, too. I feel like the animation is triggered after reaching 100viewport height and doesn't consider the content size of the slide, maybe this could be fixed easily, but I can't find the code for that animation :( 
Do you have any pointers for me where to find the code for the slide animation?

File: src/rust-api-telemetry-workshop.njk [hunk 12]
```diff
@@ -0,0 +1,239 @@
...
+      </section>
+
+      <section class="lp-section container container--lg">
+        {% set 'content' = {
+          "text": "Excellent workshop! It should be noted that it requires a solid base level of Rust knowledge [...]. Here I was able to fully use the time and knowledge of the trainer! 👍",
+          "source": "Attendee of Luca's workshop held at EuroRust 2023",
+          "loading": "lazy"
+        } %}
+        {{ quote(content) }}
+      </section>
```
Comment (LukeMathWalker) [inline src/rust-api-telemetry-workshop.njk]:
```suggestion
```

**Comment by LukeMathWalker** [comment]
> @LukeMathWalker Regarding the Navigation: Maybe we should consider a site-wide sticky navigation, because I feel like it would be weird to only have a sticky nav on that particular workshop page. As a quick solution I added a CTA button right below the intro text. But I think it would be nice to have a global sticky navigation. What do you think @marcoow? Of course the Logo and the Button would have to be adjusted depending on the background colour, so that may take me a while to figure a nice solution

That's a good point, let's privilege consistency here.
Should we mitigate the issue then by adding a CTA button at the end of the "Additional information" section, where we now have the quote we want to remove?

> Do you have any pointers for me where to find the code for the slide animation?

Not my area of expertise I'm afraid 😞

**Comment by LukeMathWalker** [comment]
On the slides: can we work around the problem by having another section like the "Additional information" one which lists the topic, as it currently appears in the workshop offering page and the EuroRust page?

**Comment by marcoow** [comment]
> As a quick solution I added a CTA button right below the intro text. But I think it would be nice to have a global sticky navigation. What do you think @marcoow?

I don't see the CTA button – let's add one in the header right below the dates maybe?

I'm not sure we should have a side-wide sticky navigation – wouldn't actually be valuable on any of the other pages I guess?

**Comment by LukeMathWalker** [comment]
> I'm not sure we should have a side-wide sticky navigation – wouldn't actually be valuable on any of the other pages I guess?

Having the `Contact` CTA visible would be quite positive on most pages imo.

**Comment by marcoow** [comment]
@yaigrie @LukeMathWalker is this good to go?

**Comment by LukeMathWalker** [comment]
Not yet @marcoow 

I think we should resolve these three points before publishing:
- On the slides: can we work around the problem by having another section like the "Additional information" one which lists the topic, as it currently appears in the workshop offering page and the EuroRust page? The page is currently way too long considering how little content it actually has inside.
- Let's add the CTA at the end of the "Additional Information" section. I'd prefer the CTA to be clearly distinguished from the text (centered with respect to the page?), but open to everything here.
- Vertical spacing is still an issue on desktop from what I can see from the preview
immagine

**Comment by marcoow** [comment]
Ok, can we get this resolved today @yaigrie? Generally I'd say though let's not have this blocked by non-necessary changes, e.g. the slides thing seems like sth. we could improve afterwards as well?

**Comment by yaigrie** [comment]
@LukeMathWalker There should be less white space in the header now! Sorry, I misread it the first time and thought you only meant the spacing between the abstract and the quote!
Also I added the list instead of slides.
The button is now below the abstract and below the additional information

File: src/rust-application-telemetry-workshop.njk [hunk 13]
```diff
@@ -0,0 +1,235 @@
...
+          <li class="workshop-topics__item">
+            <div class="workshop-topics__text-wrapper">
+              <h3 class="workshop-topics__item-heading h4">Metrics</h3>
+              <p class="text--sm">Structured logs are important, but they don’t tell the full story. We will look at how to capture metric data using the metrics library, as a tool for designing alarms as well troubleshooting faulty behaviour.</p>
+            </div>
+          </li>
+        </ol>
+      </section>
+
+      <section class="color-hero color-hero--purple" data-background-color="purple">
```
Comment (LukeMathWalker) [inline src/rust-application-telemetry-workshop.njk]:
```suggestion
      <section class="color-hero color-hero--purple hero-rust" data-background-color="purple">
```

**Comment by LukeMathWalker** [comment]
Left one last suggestion, but good to go for me otherwise 👍🏻 
Thanks for all the work on this @yaigrie (and sorry for the number of revisions 🙈 )

**Review by marcoow — APPROVED** [review]

**Pull request merged**